### ISSUE TITLE
TFLite sigmoid returns NaN for very large finite inputs while Keras saturates to 0 or 1

### ISSUE THREAD

**Issue by ALinrunrun** [tensorflow/tensorflow#118391]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

2.21.0

### Custom code

Yes

### OS platform and distribution

Linux Ubuntu 22.04

### Python version

3.11

### Current behavior?

TensorFlow Lite produces different results from Keras for tf.nn.sigmoid on very large finite float32 inputs.

For large positive inputs, Keras saturates to 1.0. For large negative inputs, Keras saturates to 0.0. However, the converted TFLite model returns NaN for the largest positive and negative finite inputs.

In the reproducer below, Keras returns 1.0 for 5e29 and 0.0 for -5e29, while TFLite returns NaN for both cases.

Expected behavior?

The converted TFLite model should match Keras behavior for tf.nn.sigmoid on finite inputs. Very large positive inputs should saturate to 1.0, and very large negative inputs should saturate to 0.0, rather than returning NaN.

### Standalone code to reproduce the issue

```shell
import os
os.environ["TF_CPP_MIN_LOG_LEVEL"] = "3"

import numpy as np
import tensorflow as tf


class M(tf.keras.Model):
    def call(self, x):
        return tf.nn.sigmoid(x)


model = M()
model.build((None, 4))

tflite_model = tf.lite.TFLiteConverter.from_keras_model(model).convert()


def run(x_arr):
    keras_out = model(x_arr).numpy()

    interpreter = tf.lite.Interpreter(model_content=tflite_model)
    interpreter.allocate_tensors()
    interpreter.set_tensor(interpreter.get_input_details()[0]["index"], x_arr)
    interpreter.invoke()

    tflite_out = interpreter.get_tensor(interpreter.get_output_details()[0]["index"])
    return keras_out, tflite_out


x_pos = np.array([[5e29, 1e25, 1e15, 50.0]], dtype=np.float32)
x_neg = np.array([[-5e29, -1e25, -1e15, -50.0]], dtype=np.float32)

e_pos, t_pos = run(x_pos)
e_neg, t_neg = run(x_neg)

print("pos eager :", e_pos)
print("pos tflite:", t_pos)
print("neg eager :", e_neg)
print("neg tflite:", t_neg)

if np.allclose(e_pos, 1.0) and np.isclose(e_neg[0, 0], 0.0) and np.isnan(t_pos[0, 0]) and np.isnan(t_neg[0, 0]):
    print("BUG REPRODUCED: TFLite sigmoid returns NaN for large finite inputs")
else:
    print("not reproduced")
```

### Relevant log output

```shell
pos eager : [[1. 1. 1. 1.]]
pos tflite: [[nan 1. 1. 1.]]
neg eager : [[0.0000000e+00 0.0000000e+00 0.0000000e+00 1.9287499e-22]]
neg tflite: [[ nan 0.0000000e+00 0.0000000e+00 1.9287498e-22]]
BUG REPRODUCED: TFLite sigmoid returns NaN for large finite inputs
```

**Comment by Kinetic-Labs-GT** [comment]
Hello everyone,

I have submitted a fix for this issue in PR #119354. I have implemented an input-clamping mechanism that ensures the Logistic/Sigmoid activation saturates correctly at the boundaries for extreme finite inputs, preventing the NaN results currently observed.

The PR currently has all the necessary logic, including a new regression unit test to ensure stability against future optimizations. I am currently finalizing the CLA signature sync to clear the automated checks.

Thank you for your patience—I look forward to your feedback on the implementation.

**Comment by aravindmurali22** [comment]
Hi @ALinrunrun ,

Thank you for filing the issue. I wasn't able to reproduce the bug using the provided script, it's returning the expected output on my side. It looks like the issue might be resolved. Could you try running it again and let me know your results?

logs: 
```
pos eager : [[1. 1. 1. 1.]]
pos tflite: [[1. 1. 1. 1.]]
neg eager : [[0.0000000e+00 0.0000000e+00 0.0000000e+00 1.9287498e-22]]
neg tflite: [[0.00000e+00 0.00000e+00 0.00000e+00 1.92875e-22]]
not reproduced
```

**Comment by aravindmurali22** [comment]
Hi @ALinrunrun ,

We verified this on our end with TensorFlow 2.21.0 (Python 3.14, NumPy 2.0.2, Keras 3.10.0, Flatbuffers 25.12.19, Absl-py 2.3.1, Requests 2.32.5) and the issue appears to be resolved, as both eager and TFLite outputs now correctly match. Thank you again!
